File: July 2024/16-07-2024/Remaining String.js
//{ Driver Code Starts
// Initial Template for javascript

'use strict';

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', inputStdin => { inputString += inputStdin; });

process.stdin.on('end', _ => {
    inputString =
        inputString.trim().split('\n').map(string => { return string.trim(); });

    main();
});

function readLine() { return inputString[currentLine++]; }

function main() {
    let t = parseInt(readLine());
    let i = 0;
    for (; i < t; i++) {
        let S = readLine().trim();
        let ch = readLine().trim();
        let count = parseInt(readLine());
        let obj = new Solution();
        let res = obj.printString(S, ch, count);
        console.log(res)
    }
}

// } Driver Code Ends


// User function Template for javascript

/**
 * @param {string} S
 * @param {string} ch
 * @param {number} count
 * @return {string}
 */

class Solution {
    printString(s, ch, count) {
     //Your Code Start
            let occ= 0;
    
    for (let i = 0; i < s.length; i++) {
        if (s[i] === ch) {
            occ++;
            if (occ === count) {
                return s.substring(i + 1);
            }
        }
    }
    
    return "";
    //Your Code End
    }
}

File: July 2024/16-07-2024/Remaining String.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;


// } Driver Code Ends
// User function template for C++
class Solution {
  public:

    string printString(string s, char ch, int count) {
      //Your Code Start
            int occ = 0;
    
    for (size_t i = 0; i < s.length(); ++i) {
        if (s[i] == ch) {
            occ++;
            if (occ == count) {
                return s.substr(i + 1);
            }
        }
    }
    
    return "";
    //Your Code End
    }
};

//{ Driver Code Starts.

int main() {

    ios_base::sync_with_stdio(0);
    cin.tie(NULL);
    cout.tie(NULL);

    int t;
    cin >> t;
    while (t--) {
        string s;
        char ch;
        int count;

        cin >> s >> ch >> count;
        Solution ob;
        cout << ob.printString(s, ch, count) << "\n";
    }

    return 0;
}
// } Driver Code Ends
